src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2477,3 +2484,353 @@ def _aks_mesh_update(
+    return True
+
+def aks_mesh_check_mesh_proxy_config(cmd, client, resource_group_name, name, v1):
+    from pprint import pprint
+    cm = v1.list_config_map_for_all_namespaces()
+    mesh_config = {}
+    root_name_space = ''
+    istio_version = ''
+    proxy_config = {}
+    
+    for item in cm.items:
+        if (item.data and 'mesh' in item.data):
+            mesh_config = yaml.safe_load(item.data['mesh'])
+            try:
+                #NOTE: This label contains version info only on non-helm installation of OSS Istio
+                istio_version = item.metadata.labels['operator.istio.io/version']
+            except Exception:
+                istio_version = ""
+    #Istio version is stored under global values tag in case of helm installation
+    if istio_version == "":
+        for item in cm.items:
+            try:
+                istio_version = yaml.safe_load(item.data['values'])['global']['tag']
+            except Exception:
+                continue
+    
+    #MeshConfig allowlist only contains supported ASM versions
+    compatible_version = True
+    if istio_version not in MESH_CONFIG_ALLOW_LIST.keys():
+        compatible_version = False
+        result_dict['Istio version'] = {'status': False}
+        supported_versions = [value for value in MESH_CONFIG_ALLOW_LIST.keys()]
+        check_istio_version(supported_versions, istio_version, result_dict)
+    else:
+        result_dict["Istio version"] = {
+            'status': True,
+            'message': "Supported Istio version found."
+        }
+    
+    if len(mesh_config) != 0:
+        proxy_config = mesh_config['defaultConfig']
+    if compatible_version:
+        check_config(mesh_config, MESH_CONFIG_ALLOW_LIST, istio_version, "Meshconfig")
+        check_config(proxy_config, PROXY_CONFIG_ALLOW_LIST, istio_version, "Proxyconfig")
+    return [istio_version, compatible_version]
+    
+
+def aks_mesh_check_base_requirements(cmd, client, resource_group_name, name, v1):
+    from pprint import pprint
+    KUBERNETES_VERSION = aks_show(cmd, client, resource_group_name, name).kubernetes_version
+    # KUBERNETES_VERSION = show_result.kubernetes_version
+    
+    if (version.parse(KUBERNETES_VERSION) < version.parse('1.24.9')):
+        result_dict['AKS version'] = {
+            'status': False,
+            'message': "Unsupported kubernetes version found."
+        }
+    else:
+        result_dict['AKS version'] = {
+            'status': True,
+            'message': "Supported Kubernetes version found."
+        }
+    try:
+        mc = client.get(resource_group_name, name)
+    except Exception as ex:
+        raise CLIError("failed to get cluster, error: %s", ex)
+
+
+    if (mc.addon_profiles and CONST_OPEN_SERVICE_MESH_ADDON_NAME in mc.addon_profiles and
+    mc.addon_profiles[CONST_OPEN_SERVICE_MESH_ADDON_NAME].enabled):
+        result_dict['OSM installation'] = {
+            'status': False,
+            'message': "OSM installation found."
```

Comment:
I think this would give the user more explicit guidance:
```suggestion
            'message': "OSM add-on must be disabled before migration to the Istio-based service mesh add-on."
```